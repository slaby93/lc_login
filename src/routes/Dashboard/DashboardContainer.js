import React from 'react';
import { connect } from 'react-redux';
import { USERS_LOGOUT } from 'ducks/user';
import Dashboard from './Dashboard';

export const DashboardContainer = ({ onLogout }) => {
  return (
    <Dashboard
      onLogout={onLogout}
    />
  );
};

export function mapDispatchToProps(dispatch) {
  return {
    onLogout: () => {
      dispatch(USERS_LOGOUT());
    },
  };
}

export default connect(null, mapDispatchToProps)(DashboardContainer);
