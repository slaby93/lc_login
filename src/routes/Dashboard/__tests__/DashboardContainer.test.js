import React from 'react';
import { shallow } from 'enzyme';
import { DashboardContainer, mapDispatchToProps } from './../DashboardContainer';

jest.mock('./../Dashboard');

describe('<DashboardContainer />', () => {
  test('basic render', () => {
    const mountedComponent = shallow(<DashboardContainer />);
    expect(mountedComponent).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    test('result shape', () => {
      dispatch.mockReset();
      const result = mapDispatchToProps(dispatch);
      expect(result).toMatchSnapshot();
    });

    test('onLogout', () => {
      dispatch.mockReset();
      const result = mapDispatchToProps(dispatch);
      result.onLogout();
      expect(dispatch).toBeCalledWith({ type: 'USER/LOGOUT' });
    });
  });
});
