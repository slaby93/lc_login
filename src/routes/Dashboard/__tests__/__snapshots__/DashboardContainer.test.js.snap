// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardContainer /> basic render 1`] = `<Dashboard__StyledComponent />`;

exports[`<DashboardContainer /> mapDispatchToProps result shape 1`] = `
Object {
  "onLogout": [Function],
}
`;
